### PR TITLE
feat: Add helper functions for Tailwind support: shorthand expansion and variable substitution

### PR DESCRIPTION
**Pull request by istarkov**
## Description

Gonna merge this before next step.

Helper functions to parse tailwind.

1. `expandTailwindShorthand` - Expands shorthand CSS properties into their long-form equivalents. _'mx' is expanded to 'mr' (margin-right) and 'ml' (margin-left)._
2. `substituteVariables` - Resolves CSS variables by replacing their references with actual values. _Transforms `.class { --property: 1px; margin: var(--property); }` into `.class { margin: 1px; }`._

## Steps for reproduction

See tests.

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: packages/css-data/src/tailwind-parser/shorthand.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  "rounded-r": ["rounded-tr", "rounded-br"],
+  "rounded-b": ["rounded-br", "rounded-bl"],
+  "rounded-l": ["rounded-tl", "rounded-bl"],
+
+  "border-spacing": ["border-spacing-x", "border-spacing-y"],
+
+  "border-w": ["border-w-t", "border-w-r", "border-w-b", "border-w-l"],
+  "border-w-x": ["border-w-r", "border-w-l"],
+  "border-w-y": ["border-w-t", "border-w-b"],
+  "border-color": [
```
[the remark this answers is not in the document]
Comment (istarkov) [reply packages/css-data/src/tailwind-parser/shorthand.ts:67]: Checking

File: packages/css-data/src/tailwind-parser/shorthand.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+/**
+ * A little bit changed tailwind-merge
+ * conflictingClassGroups https://github.com/dcastil/tailwind-merge/blob/fb1fc1f9d106f2184e24949e6e6d1a67904706fa/src/lib/default-config.ts#L1716
+ *
+ * Attributions
+ * MIT License
+ * Copyright (c) 2021 Dany Castillo
+ **/
+const conflictingClassGroups: Record<string, string[]> = {
```
Comment (kof) [inline packages/css-data/src/tailwind-parser/shorthand.ts:35]: is there a chance tailwind will evolve and we would have to keep up with this function? wish we would take what they use

File: packages/css-data/src/tailwind-parser/substitute.test.ts [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+*,
+::before,
+::after {
+ --var-a: rgba(111, 111, 111, 0.5);
+ --var-b: rgba(222, 222, 222, 0.5);
+}
+`;
+
+describe("expandTailwindShorthand", () => {
+  it("resolves simple variable", () => {
```
Comment (kof) [inline packages/css-data/src/tailwind-parser/substitute.test.ts:15]: we are using test() everywhere

File: packages/css-data/src/tailwind-parser/substitute.ts [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+   * Custom properties treat variables as strings, making direct resolution impossible.
+   * We perform multiple iterations to substitute these variables with their actual values.
+   *
+   * Example:
+   * Initial: .class { --var: var(--margin-y) var(--margin-x); margin: var(--var); }
+   * Iteration 1: .class { margin: var(--margin-y) var(--margin-x); }
+   * Iteration 2: .class { margin: 1px 2px; }
+   */
+  const MAX_NESTED_DEPENDENCIES_DEPTH = 5;
+  for (let i = 0; i !== MAX_NESTED_DEPENDENCIES_DEPTH; ++i) {
```
Comment (kof) [inline packages/css-data/src/tailwind-parser/substitute.ts:52]: please don't set an example with single-char variables

File: packages/css-data/src/tailwind-parser/substitute.test.ts (hunk 3 above)
Comment (istarkov) [reply packages/css-data/src/tailwind-parser/substitute.test.ts:15]: `test` doesn't sound with almost any text

**Review by kof — APPROVED** [review]

File: packages/css-data/src/tailwind-parser/shorthand.ts (hunk 2 above)
Comment (istarkov) [reply packages/css-data/src/tailwind-parser/shorthand.ts:35]: Its our internal details that we don't support shorthands

File: packages/css-data/src/tailwind-parser/shorthand.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (istarkov) [reply packages/css-data/src/tailwind-parser/shorthand.ts:69]: Removed border-color for now as we are using shorthand

File: packages/css-data/src/tailwind-parser/substitute.test.ts (hunk 3 above)
Comment (kof) [reply packages/css-data/src/tailwind-parser/substitute.test.ts:15]: its just not bdd, but its possible to write test() in a way that is well-worded

**Pull request merged**